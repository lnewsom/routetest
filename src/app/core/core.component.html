<app-header [coreId]="this.coreId"></app-header>
<router-outlet></router-outlet>
