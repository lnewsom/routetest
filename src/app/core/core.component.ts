import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-core',
  templateUrl: './core.component.html',
  styleUrls: ['./core.component.scss']
})
export class CoreComponent implements OnInit {

  constructor(private route: ActivatedRoute) { }

  ngOnInit() {
    this.route.parent.params.subscribe((params) => {
      console.log('Core Params: ' , params);
    });
  }

}
