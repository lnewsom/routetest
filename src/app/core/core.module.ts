import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CoreComponent } from './core.component';
import { HeaderComponent } from './header/header.component';
import { AppRoutingModule } from '../app-routing.module';


@NgModule({
  declarations: [
    CoreComponent,
    HeaderComponent
  ],
  imports: [
    CommonModule,
    AppRoutingModule
    ], exports: [
    HeaderComponent
  ]
})
export class CoreModule { }
