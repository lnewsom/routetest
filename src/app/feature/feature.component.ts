import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-feature',
  templateUrl: './feature.component.html',
  styleUrls: ['./feature.component.scss']
})
export class FeatureComponent implements OnInit {

  constructor(private route: ActivatedRoute) { }

  ngOnInit() {
    this.route.parent.params.subscribe((params) => {
      console.log('Feature Params: ' , params);
    });
  }

}
