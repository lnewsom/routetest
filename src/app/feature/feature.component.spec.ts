import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { FeatureComponent } from './feature.component';
import { ActivatedRoute } from '@angular/router';

jest.mock('@angular/router');
describe('FeatureComponent', () => {

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      providers: [ActivatedRoute],
      declarations: [FeatureComponent]
    }).compileComponents();
  }));

  test('snapshot matches', () => {
    const fixture = TestBed.createComponent(FeatureComponent);
    expect(fixture).toMatchSnapshot();
  });
});
