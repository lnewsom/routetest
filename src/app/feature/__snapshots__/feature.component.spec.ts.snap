// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureComponent snapshot matches 1`] = `
<app-feature
  route={[Function ActivatedRoute]}
>
  <p>
     feature works!
  
  </p>
</app-feature>
`;
